src/encoded/types/biosample.py

```diff
@@ -553,125 +756,101 @@ def summary(self, request,
+                suffix = ''
             else:
-                if len(dict_of_phrases['rnais']) > 1:
-                    summary_phrase += dict_of_phrases['transfection'] + ' RNAi knockdown ' + \
-                        ', '.join(map(str, dict_of_phrases['rnais']))
-
-        if 'treatments' in dict_of_phrases:
-            if len(dict_of_phrases['treatments']) == 1:
-                summary_phrase += ' treated with '+dict_of_phrases['treatments'][0]
+                suffix = generate_sentence(phrases_dictionarys[0], sentence_parts[max_index+1:])
+            middle = []
+            for d in phrases_dictionarys:
+                part_to_add = generate_sentence(d, sentence_parts[min_index:max_index+1])
+                if part_to_add != '':
+                    middle.append(part_to_add)
+                else:
+                    constructed_middle = []
+                    for x in range(min_index, max_index+1):
+                        if sentence_parts[x] in negations_dict:
+                            constructed_middle.append(negations_dict[sentence_parts[x]])
+                    if len(constructed_middle) == 0:
+                        middle.append('NONE')
+                    elif len(constructed_middle) == 1:
+                        middle.append(constructed_middle[0])
+                    else:
+                        middle.append(', '.join(map(str, constructed_middle)))
+            sentence_middle = sorted(list(set(middle)))
+            if prefix == '':
+                sentence_to_return = ' and '.join(map(str, sentence_middle))
             else:
-                if len(dict_of_phrases['treatments']) > 1:
-                    summary_phrase += ' treated with ' + \
-                                      ', '.join(map(str, dict_of_phrases['treatments']))
-
-        # need to rephrase later on I guess
-        if 'depleted_in' in dict_of_phrases:
-            summary_phrase += ' ' + dict_of_phrases['depleted_in']
-
-        if 'talens' in dict_of_phrases:
-            summary_phrase += ' ' + dict_of_phrases['talens']
-
-        if 'constructs' in dict_of_phrases:
-            summary_phrase += ' ' + dict_of_phrases['constructs']
-        if 'model_organism_constructs' in dict_of_phrases:
-            summary_phrase += ' ' + dict_of_phrases['model_organism_constructs']
-
-        if summary_phrase.strip()[-1] == ';':
-            return summary_phrase.strip()[:-1]  # , str(dict_of_phrases))
-        else:
-            return summary_phrase.strip()
+                sentence_to_return = prefix.strip() + ' ' + \
+                    ' and '.join(map(str, sentence_middle))
+            if suffix != '':
+                sentence_to_return += ', ' + suffix
+    else:
+        sentence_to_return = generate_sentence(phrases_dictionarys[0], sentence_parts)
+
+    sentence_to_return = sentence_to_return.replace(' percent', '%')
+    sentence_to_return = sentence_to_return.replace('1 hours', '1 hour')
```

Comment:
Do you mean 'percent > %' or stripping the colon at the end? I am not sure what can be done using grammar modules?

I am failing so far to find any defined sentence structure/format I can try to enforce on the summaries.